src/js/client.ts

```diff
@@ -167,12 +167,13 @@ export class ReactNativeClient extends BaseClient<ReactNativeClientOptions> {
       didCallNativeInit = await NATIVE.initNativeSdk(this._options);
     } catch (_) {
       this._showCannotConnectDialog();
-
-      this._options.onReady?.({ didCallNativeInit: false });
-
-      return;
+    } finally {
+      try {
+        this._options.onReady?.({ didCallNativeInit });
+      } catch (error) {
+        logger.error('The OnReady callback threw an error: ', error);
+      }
```

Comment:
The logic here changed or am I missing something? at least `didCallNativeInit` isn't always `false` as before.

Reply:
yeah, I guess its cleaner and has the same effect, happy with it.